### PR TITLE
ONYX-13637:Add more UT for claim name validation

### PR DESCRIPTION
**Pull request by semyon-estrin**
### Desired Outcome

Add more UT for claim name validation.
Added tests for not supported characters `* - % ! ( & @ ^ ~ \ + =` 
Added test for supported character `|`

### Implemented Changes

Added 13 UT for claim name input validation.

### Connected Issue/Story

Additional test cases, 
Related to Internal CyberArk issue [ONYX-13637](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-13637)

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Review by sashaCher — COMMENTED** [review]
'|' is looking like bug
it shouldn't be a valid character

File: spec/app/domain/authentication/authn-jwt/input_validation/validate_claim_name_spec.rb [hunk 1]
```diff
@@ -455,4 +611,4 @@
       end
     end
   end
-end
+end
```
Comment (tzheleznyak) [inline spec/app/domain/authentication/authn-jwt/input_validation/validate_claim_name_spec.rb:614]: Add line

**Comment by sashaCher** [comment]
@semyon-estrin @tzheleznyak 
Do you think it worth to simplify us in the file with similar to https://github.com/cyberark/conjur/blob/master/spec/app/domain/authentication/authn-jwt/restriction_validation/validate_restriction_name_spec.rb approach?

**Comment by semyon-estrin** [comment]
> @semyon-estrin @tzheleznyak Do you think it worth to simplify us in the file with similar to https://github.com/cyberark/conjur/blob/master/spec/app/domain/authentication/authn-jwt/restriction_validation/validate_restriction_name_spec.rb approach?

Refactored according to this approach. Please re-review.

**Review by tzheleznyak — APPROVED** [review]
LGTM

**Pull request merged**